### ISSUE TITLE
Only one proxy used out of the whole list?

### ISSUE THREAD

**Issue by aquanetworks** [modrzew/pokeminer#217]
We find that only 1 LAST proxy is used from the array of proxies.

This is confirmed by running "netstat -anp | grep PROXY_PORT".

Is this a bug or known behavior that pokeminer would use only 1 proxy until it gets banned/goes down?

**Comment by Aiyubi** [comment]
you misunderstand the proxy list. Request (the module used) only supports 1 proxy and also the structure to save the data for proxy only has 1 field "http" that you overwrite if you define it new.

However: I programmed myself some way to use different proxy for different accounts. Will add a PR once the next release is out.

**Comment by gunawanputra** [comment]
What about proxy jump? Create a proxy dictionary just like the account, then detect each worker that get banned and set banned workers connection to other proxy.
 
I'm already create a banned detection, by checking forts lenght on each scan. Once it's counting reach 5 times for example, banned worker thread will be shutdown, not just killed and restart. If using proxy then it could modified to reroute the connection instead of getting shutdown.

On my scanning, I found out that at some moment not all workers are getting banned, that's what make me create banned detection.

I'm still waiting for next release, then will try to add a PR about banned detection if it's not exist yet.

**Comment by Aiyubi** [comment]
you are not a python dev?
dictionary={}
list=[]
tuple=()
so accounts is a list of tuples, not adictionary like the proxies (you are mixing up words)

I think your request is very special, but if you wanna create a PR we will see if it gets accepted

**Comment by gunawanputra** [comment]
Yes I'm new to python. What I mean is create a list of proxy dictionary, then when a worker gets banned we just need to grab another proxy address from the list and start a new connection.

**Comment by Aiyubi** [comment]
okay then we have 2 usecases proxys are for:
* people who need proxys to access internet (I recomment using the proxy variable as default)
* people who want to route through different proxys to balance requests per IP.

When designing this I recommend not using the same variable for both since those are different usecases.

**Comment by modrzew** [comment]
What @gunawanputra says makes sense. Though I'm not that much inclined to do it, since my accounts are still working without need for proxy :wink:

If anyone wants to give it a try, here's a hint: [worker.pyL87](https://github.com/modrzew/pokeminer/blob/53d80dd302d89cb7229420eab4c0fe7037a8db36/worker.py#L87). Instead of just using `config.PROXY`, write something that will keep state of all proxies and provide the best one to freshly-created slaves.

**Comment by and7ey** [comment]
> However: I programmed myself some way to use different proxy for different accounts. Will add a PR once the next release is out.
@Aiyubi , could you please share your code to use multiple proxies?

I wanted to run pokeminer on AWS, but AWS's IPs are banned by Niantic.
